doc/sphinx/docs/python/tutorials/hypervolume.rst

```diff
@@ -0,0 +1,180 @@
+
+.. image:: ../../images/hv_front_2d_simple.png
+  :width: 600px
+  :align: right
+
+Once the hypervolume object is created, it allows for the computation of the following figures:
+
+1. :class:`~pygmo.core.hypervolume.compute` - Returns the joint hypervolume of the set of points (S-Metric).
+
+.. doctest::
+
+    >>> # hv and ref_point refer to the data above
+    >>> hv.compute(ref_point)
+    3.25
+
+2. :class:`~pygmo.core.hypervolume.exclusive` - Returns the exclusive hypervolume by the point at given index.
+   The exclusive hypervolume 
+   is defined as the part of the space dominated exclusively by one point and is also called its (hypervolume) contribution.
+
+.. doctest::
+
+    >>> # hv and ref_point refer to the data above
+    >>> hv.exclusive(1, ref_point)
+    0.25
+    >>> hv.exclusive(3, ref_point)
+    0.0
+
+3. :class:`~pygmo.core.hypervolume.least_contributor` - Returns the index of a point contributing the least to the hypervolume.
+
+.. doctest::
+
+    >>> # hv and ref_point refer to the data above
+    >>> hv.least_contributor(ref_point)
+    3
+
+4. :class:`~pygmo.core.hypervolume.greatest_contributor` - Returns the index of a point contributing the most to the hypervolume.
+
+.. doctest::
+
+    >>> # hv and ref_point refer to the data above
+    >>> hv.greatest_contributor(ref_point) # doctest: +SKIP
+    0
+
+.. note::
+  In case of several least/greatest contributors, pygmo returns only one contributor out of all candidates arbitrarily.
+
+5. :class:`~pygmo.core.hypervolume.contributions` - Returns a list of contributions for all points in the set.
+   This returns the same results as the successive call to the :class:`~pygmo.core.hypervolume.exclusive` method 
+   for each of the points. Due to the implementation, calling :class:`~pygmo.core.hypervolume.contributions` once can
+   be much faster (up to a linear factor) than computing all contributions separately 
+   by using :class:`~pygmo.core.hypervolume.exclusive`.
+
+.. doctest::
+
+  >>> # hv and ref_point refer to the data above
+  >>> hv.contributions(ref_point) 
+  array([ 0.5 ,  0.25,  0.5 ,  0.  ])
+
+Since all of the methods above require a reference point, it is useful to generate
+one automatically using the :class:`~pygmo.core.nadir`
+
+This following short script makes use of some of the features mentioned above to show the hypervolume
+increase after the evolution of a :class:`~pygmo.core.population`.
+
+.. doctest::
+
+    >>> import pygmo as pg
+    >>> # Instantiates a 4-objectives problem
+    >>> prob = pg.problem(pg.dtlz(id=4, dim = 12, fdim=4))
+    >>> pop = pg.population(prob, 84)
+    >>> # Construct the hypervolume object
+    >>> # and get the reference point off-setted by 10 in each objective
+    >>> hv = pg.hypervolume(pop)
+    >>> offset = 5
+    >>> ref_point = [max(pop.get_f(), key = lambda x: x[it])[it] + offset for it in [0,1,2,3]]
+    >>> hv.compute(ref_point) # doctest: +SKIP
```

Comment:
:) thanks, yes I used the ellipsis syntax before, but in this case the problem is that the whole result may be completely different as the ref_point will be kind of random